### ISSUE TITLE
REST API returns 404

### ISSUE THREAD

**Issue by Sergio-** [back2arie/Kalkun#209]
Hello Folks. Good day.

I'm using Kalkun 0.7.1 from Github. Web interface works fine. I then enabled REST API plugin and when trying to use it, I got a 404 error.
I'm using the Master branch as of Apr, 19th 2020. I read other issues related to it and on the comments, it seems it was resolved on #200, however, I'm still facing the issue, here's a sample URL I'm using:
http://192.168.10.81/Kalkun-master/index.php/plugin/rest_api/send_sms?phoneNumber=994867XXX&message=testing

My main Kalkun instante is on: http://192.168.10.81/Kalkun-master/index.php (working fine).

Appreciate any pointers to help solve the problem

Thank you.

**Comment by tenzap** [comment]
#200 has been merged into "devel"
Can you try again but with latest devel instead of "master" ?

**Comment by Sergio-** [comment]
Thanks for the quick response.
I tested with the Devel branch and the 404 problem is gone. Calling the REST API returns a nice XML stating the message is queued.
However, I'm not sure why... But the messages created with the REST API never leaves the Outbox. Everything else seems to be working fine but the REST API message shows up in the Outbox indefinitely...
Any idea on how to actually make the REST API message to be processed?
Thank you.

**Comment by tenzap** [comment]
I just compared 2 SMS directly in the outbox table of the database (1 with REST, the other created through the GUI)
The only difference was that the message created by REST api has a sender ID of value "false"
Deleting "false" directly on the table makes the message go to outbox

**Comment by tenzap** [comment]
[Gammu doc](https://wammu.eu/docs/manual/smsd/tables.html#outbox)  states : 
```
SenderID (text)
    which SMSD instance should send this one sequence, see PhoneID and Multiple modems.
If blank, first SMSD who sees this message first will process it.
```

**Comment by tenzap** [comment]
Should be fixed once #212 is merged.
In the meantime, you can add this parameter to your URL and leave its value empty
`&SenderID=`
--> 
http://192.168.10.81/Kalkun-master/index.php/plugin/rest_api/send_sms?phoneNumber=994867XXX&message=testing&SenderID=

**Comment by Sergio-** [comment]
Thanks for the quick response.
Perfect. Adding the SenderID param in the URL does the trick. It is working as expected now.
I'll close this issue.
Thank you.

**Comment by tenzap** [comment]
Could you just confirm that #212 fixes it too (no need to set &SenderID=) ?
I has been merged into devel in the meantime
Thank you

**Comment by Sergio-** [comment]
Yes! It does work without the sender ID using the most recent devel version.
Thank for fixing it.
On a side note.... How can I send a message with special characters (ã, é, ç, etc) using the REST API?
Thank you.

**Comment by tenzap** [comment]
Thanks

Good question...

**Comment by tenzap** [comment]
Just add the text in the message ?
I didn't face an issue for sending through firefox

**Comment by Sergio-** [comment]
Thank you.
That's interesting. It actually works partially. For example: If I use a &message=ção, the SMS is sent, but the special characters are stripped out. I receive cao;
However, If I send &message=Éé, I receive then correctly in the SMS (Éé).
Any ideas why it works for some special chars and others not?
Thank you.

**Comment by tenzap** [comment]
I don't really know. It could be (or maybe not ? ;) ) a configuration problem on any system between your client & Kalkun (your http client, http server, php, kalkun, the way you do the query) I would check the charset/encoding config of each system in the chain.

I'm under Linux/Debian and I suppose everything is UTF-8. Even if I put emojis in the url they are correctly set in the SMS of the Kalkun queue...

**Comment by Sergio-** [comment]
Yes, I'm investigating it too, Althought, I could not find any concrete reason yet. I tested it via Browser URL in Chromium and Firefox, both updated to the last version in both Ubuntu 18.04 and Fedora 31. Same behaviour, I also tested ir via CURL terminal on both systems, same behaviour... I'm sort of clueless at this point. :-\
Thank you.

**Comment by Sergio-** [comment]
I believe I found some clue... 
It seems the messages sent by the REST API are not being marked as Unicode, looking at the DB Record, the characters are correctly displayed in the DB record, however the "Coding" field contains "Default_No_Compression" instead of "Unicode_No_Compression".
Anyway to change or force the REST API messages to be marked as Unicode?
Thank you.

**Comment by tenzap** [comment]
Are you using a mysql database ? What is the encoding of the tables you mention ? In the SQL scripts for mysql it is often "latin1" as charset. Could that be the reason ?
I'm running Postgresql (and the db has ENCODING = 'UTF8')

If you think the issue is related to kalkun, could you open a new issue ?

**Comment by Sergio-** [comment]
Yes, It does make sense to move this last discussion to a different Issue.
I opened issue #213 for it.
Thank you.
